tests/Oryx.Integration.Tests/Php/PhpFpmConfigTest.cs

```diff
@@ -0,0 +1,170 @@
+        [InlineData("10", null, "9", null, "8", null, "7", null, "pm.start_servers(9) must not be less than pm.min_spare_servers(7) and not greater than pm.max_spare_servers(8)")]
+        public async Task PipelineTestInvocationsPhp80Async(
+            string fpmMaxChildren, string expectedFpmMaxChildren,
+            string fpmStartServers, string expectedFpmStartServers,
+            string fpmMaxSpareServers, string expectedFpmMaxSpareServers,
+            string fpmMinSpareServers, string expectedFpmMinSpareServers,
+            string failureOutputText)
+        {
+            await PhpFpmConfigTestAsync(
+                "8.0", 
+                fpmMaxChildren, expectedFpmMaxChildren,
+                fpmStartServers, expectedFpmStartServers,
+                fpmMaxSpareServers, expectedFpmMaxSpareServers,
+                fpmMinSpareServers, expectedFpmMinSpareServers,
+                failureOutputText);
+        }
+
+        [Theory, Trait("category", "php-7.4")]
+        [InlineData("10", "10", "7", "7", "8", "8", "6", "6", null)]
+        [InlineData("", "5", "", "2", "", "3", "", "1", null)] // defaults
+        [InlineData("false", null, "5", null, "5", null, "5", null, "pm.max_children must be a positive value")]
+        [InlineData("-1", null, "5", null, "5", null, "5", null, "pm.max_children must be a positive value")]
+        [InlineData("10", null, "5", null, "13", null, "12", null, "pm.min_spare_servers(12) and pm.max_spare_servers(13) cannot be greater than pm.max_children(10)")]
+        [InlineData("10", null, "9", null, "8", null, "7", null, "pm.start_servers(9) must not be less than pm.min_spare_servers(7) and not greater than pm.max_spare_servers(8)")]
+        public async Task PipelineTestInvocationsPhp74Async(
+            string fpmMaxChildren, string expectedFpmMaxChildren,
+            string fpmStartServers, string expectedFpmStartServers,
+            string fpmMaxSpareServers, string expectedFpmMaxSpareServers,
+            string fpmMinSpareServers, string expectedFpmMinSpareServers,
+            string failureOutputText)
+        {
+            await PhpFpmConfigTestAsync(
+                "7.4",
+                fpmMaxChildren, expectedFpmMaxChildren,
+                fpmStartServers, expectedFpmStartServers,
+                fpmMaxSpareServers, expectedFpmMaxSpareServers,
+                fpmMinSpareServers, expectedFpmMinSpareServers,
+                failureOutputText);
+        }
+
+        private async Task PhpFpmConfigTestAsync(
+            string phpVersion, 
+            string fpmMaxChildren, string expectedFpmMaxChildren,
+            string fpmStartServers, string expectedFpmStartServers,
+            string fpmMaxSpareServers, string expectedFpmMaxSpareServers,
+            string fpmMinSpareServers, string expectedFpmMinSpareServers,
+            string failureOutputText)
```

Comment:
nit: I would have `failureOutputText` be optional so a test that doesn't expect a failure doesn't need to worry about providing something for this argument. I would also apply this to your tests above so that the `InlineData` sets that don't expect a failure don't have to provide a `null` value to the argument, they just don't provide anything.

```cs
[Theory]
[InlineData("foo", "FAILED")]
[InlineData("bar")]
public async Task TestSomethingAsync(
    string name,
    string failureOutputText = default)
{
    // Test
}
```